fgpyo/util/metric.py

```diff
@@ -149,7 +152,12 @@ def _parsers(cls) -> Dict[type, Callable[[str], Any]]:
         return {}
 
     @classmethod
-    def read(cls, path: Path, ignore_extra_fields: bool = True) -> Iterator[Any]:
+    def read(
+        cls,
+        path: Path,
+        ignore_extra_fields: bool = True,
+        header_comment_char: str = DEFAULT_HEADER_COMMENT_CHAR,
```

Comment:
A couple of thoughts:

1. I think this should probably be `: Optional[str] = None` for backwards compatibility
2. Should we also allow for blank lines before the header?
3. Should lines starting with the comment character (and blank lines?) be allowed in the body of the data or only pre-header?

Reply:
And one more ... since python has no `char` type, maybe this should be `comment_prefix`?  And then it could support e.g. `##` or `//`.

Reply:
1. Yep, agreed
2. This PR does allow for blank lines before the header - see line 366. Any lines that start with the comment character or are blank are skipped.
3. I don't think so. I'm not familiar with any format where that's valid, and I think it's more likely to be a sign of a malformatted file or a concatenation gone wrong.
4. Yep, agreed

I'll update (1) and (4)

Reply:
See #124